repository_pool.go

```diff
@@ -35,10 +35,47 @@ func NewRepository(id string, repo *git.Repository) *Repository {
 	}
 }
 
+// Close closes all opened files in the repository.
+func (r *Repository) Close() {
+	if r != nil && r.Repository != nil {
+		f, ok := r.Storer.(*filesystem.Storage)
+		if ok {
+			// The only type of error returned is "file already closed" and
+			// we don't want to do anything with it.
+			f.Close()
+		}
+	}
+}
+
 // NewRepositoryFromPath creates and initializes a new Repository structure
 // and initializes a go-git repository
 func NewRepositoryFromPath(id, path string) (*Repository, error) {
-	repo, err := git.PlainOpen(path)
+	op := filesystem.Options{
+		ExclusiveAccess: true,
+		KeepDescriptors: true,
+	}
+
+	var wt billy.Filesystem
+	fs := osfs.New(path)
+	f, err := fs.Stat(git.GitDirName)
```

Comment:
What would happen if  `path` is a bare repository?

Reply:
If it's a bare repo then `.git` does not exist and the path bellow (`if != nil && f.IsDir()`) not taken. This means that worktree will be nil and the storer is created with the fs provided instead of chroot into `.git`.